src/Constraint/AbstractConstraint.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Db\Constraint;
+
+/**
+ * Represents an abstract constraint in a database.
+ *
+ * A constraint is a rule that's enforced on the data in a table, such as a primary key, a foreign key, a default value,
+ * check or index constraint.
+ *
+ * It's the base class for all constraint classes, and defines methods that are common and shared by all constrained
+ * classes, with name and columnNames being the most common ones.
+ */
+abstract class AbstractConstraint
```

Comment:
Are setters needed in these classes?

May be keep only constructors with public read only properties?